### DISCUSSION TITLE
프롤로그 온보딩 문서화 설문

### DISCUSSION THREAD

**Discussion by leo0842** [woowacourse/prolog#1082] [Ideas]
안녕하세요 프롤로그 근로자 여러분!
저희는 기술개선 스쿼드의 후이, 이스트, 에덴입니다.

저희가 오늘 발표한 문서화 초안에 대한 피드백을 받고자 합니다.

설문에 참여해주신 분들 중 5분을 선정하여 **배달의 민족 1만원 쿠폰**을 드리고자 합니다.

정성과 관심이 많이 담긴 피드백 부탁드립니다.

- [모니터링](https://github.com/leo0842/prolog/blob/docs/combine-be-configuration-and-monitoring/docs/backend/%EB%AA%A8%EB%8B%88%ED%84%B0%EB%A7%81.md)
- [형상관리](https://github.com/leo0842/prolog/blob/docs/combine-be-configuration-and-monitoring/docs/backend/%ED%98%95%EC%83%81%EA%B4%80%EB%A6%AC.md)
- [인가 과정](https://github.com/woowacourse/prolog/blob/docs/combine/docs/backend/%EC%9D%B8%EA%B0%80%EA%B3%BC%EC%A0%95.md)
- [DB ERD](https://github.com/woowacourse/prolog/blob/docs/combine/docs/backend/ERD.md)
- [프론트](https://github.com/woowacourse/prolog/discussions/1069#discussioncomment-3893212)

### 1. 5기가 온보딩 기간에 해당 문서를 보았을 때 부족한 부분이 있다면 피드백 부탁드립니다.

ex)
- 형상관리: 그림도 넣어주세요.
- 프론트: 맞춤법좀 맞춰주세요.

### 2. 온보딩 기간에 위의 문서들을 제외하고 어떤 문서가 추가되면 좋을지 아이디어 부탁드립니다.

**Comment by her0807** [comment]
1. 5기가 온보딩 기간에 해당 문서를 보았을 때 부족한 부분이 있다면 피드백 부탁드립니다.
ex)

`인가과정 문서화` : 현재 AOP 로 인가 과정이 진행되고 있는데 왜 interceptor 가 아니라 , AOP 로 인가처리를 하고 있는지 .. 구현된 배경에 대한 설명을 해주셔도 좋을 것 같아요. 

`모니터링 문서화` : 현재 모니터링이 구축되어 있지 않지만, 구축의 필요성을 왜 느꼈는지 시도했던 과정도 포함되면 좋을 것 같아요. 그리고 구축하는 과정에서 기록해놨던 내용들 (슬로한테 있던 것 같은데) 그것도 추가하면 좋을 것 같아요.

`형상관리 문서화` : 여기에 PR 에 리뷰어 매칭할 때 PR 보내는 사람이 리뷰어 요청 보내고 본인이 직접 리뷰 어필해야한다 이런 꿀팁 같은 것도 자세히 남겨주면 좋을 것 같아요. 그리고 깃헙 브런치 전략 (배포 과정)도 헷갈려서 그 내용도 있으면 좋을 것 같네요 ☺️ 


2. 온보딩 기간에 위의 문서들을 제외하고 어떤 문서가 추가되면 좋을지 아이디어 부탁드립니다.
- 작업 내역이 궁금할 때 히스토리 찾아보는 법!!!(PR, 이슈, 로컬에서 담당자 이름 등등 보고, 위키 보고 ..) 
- 처음 프로젝트를 clone 받고 해야하는 설정들 (컨벤션이나, 설정파일)

**Comment by wishoon** [comment]
1. 5기가 온보딩 기간에 해당 문서를 보았을 때 부족한 부분이 있다면 피드백 부탁드립니다.
* DB ERD : 각 DB에 대해 간단한 설명이 있으면 좋을 것 같아요 (5기 분들이 조금 더 빠른 이해가 가능할 것 같습니다)

2. 온보딩 기간에 위의 문서들을 제외하고 어떤 문서가 추가되면 좋을지 아이디어 부탁드립니다.
* PROLOG 프로젝트의 컨벤션이 조금 더 정리가 되면 좋을 것 같습니다
  * 커밋 컨벤션 (PR 작성 이름, PR 템플릿)
  * 백엔드 컨벤션 (이건 희망사항인데 많은 사람들이 같이 개발을 하다보니 코드 작성방법이 조금씩 다른 것 같아서요..! 지금 개발하는 것 부터 작성 규칙이 있으면 5기가 조금 더 코드 읽기도 편할 것 같고 향후 개발에 도움이 될 것 같습니다 아마 회의가 필요한 부분일 것 같네요 ㅎㅎ ex) 빌터패턴, Lombok 사용 유뮤 등등)
* 현재 도메인 용어가 정리되어 있지 않아 부르는 호칭이 조금씩 다른데 통일된 용어 사전이 있으면 좋을 것 같습니다.
* 이벤트 스토밍을 정리해보는 것도 방법일 것 같아요. 지원 플랫폼을 보니깐 이벤트 스토밍이 잘 정리되어 있더라고요!! 지금 개발되는 내용이나, 핵심 기능 정도만 정리해도 괜찮을 것 같아요!! https://miro.com/app/board/o9J_l5s6EZI=/

**Comment by JUDONGHYEOK** [comment]
### 1. 5기가 온보딩 기간에 해당 문서를 보았을 때 부족한 부분이 있다면 피드백 부탁드립니다.

- 형상관리: 그림이 있다면 더욱 좋을 것 같습니다. 아직도 branch 전략이 헷갈리는데 구체적인 머지 전략도 추가해주시면 좋을것 같습니다.
- 프론트: 아직 정하진 않았지만 타입스크립트 관련 컨벤션도 추가되면 좋을 것 같습니다.


### 2. 온보딩 기간에 위의 문서들을 제외하고 어떤 문서가 추가되면 좋을지 아이디어 부탁드립니다.
- 아직 정의되지 않았지만 프론트엔드 코드에서 컨벤션들을 정의하고 추가되면 좋을 거 같습니다.

**Comment by euijinkk** [comment]
## 1. 5기가 온보딩 기간에 해당 문서를 보았을 때 부족한 부분이 있다면 피드백 부탁드립니다.
- 디자인 시스템 너무 좋아요!
- 컨벤션도 좋습니다. 내용 추가 제안합니다!

5기가 레벨2때 처음 만나게 될텐데요. 여기저기 다른 컨벤션을 사용하고 있어 혼동이 올 여지가 있다고 생각합니다. 그래서 사소한 부분이라도 저희끼리 논의해서 좋은 방식을 채택해서 컨벤션화 하는 것을 제안합니다. 특히, 타입스크립트과 react-query는 레벨2때 익숙하지 않을 수 있으므로 더욱 상세한 컨벤션이 필요하다고 봅니다.

### 컨벤션에 추가되었으면 하는 것
- 파일 컨벤션
  ex ) components를 만들 때 폴더 안에 styles 파일과 index 파일을 둔다.
- 스타일 컨벤션
  지금부터라도 스타일을 통일하는 것이 좋을것같습니다. css props 방식은 꼭 필요할 때만 사용하고 최대한 지양하고 걷어냈으면 좋겠습니다. 가독성을 너무 흐린다고 생각해서요! 제가 제안하는 통일안은 styles.tsx 파일을 분리하고
`import * as Styled from './styles'` 로 import 해서 `Styled.Component` 로 사용하는 방식입니다. `S.Component` 형식도 괜찮습니다.
- react-query 컨벤션
  fetcher - query/mutation hook - component 로 이어지는 흐름에서 각각을 어디 폴더에 어떤 방식으로 사용하면 좋은지 제안하면 좋을것같습니다. 타이핑도 얹으면 좋을것같네요. 
- axios instance 를 사용하는 컨벤션
  제가 `apis/index.ts` 정의해놓은 client instance를 사용하기. 
- TypeScript 컨벤션
  - model 타입들을 어디에 어떻게 선언할지. 지금도 혼재중이긴한데, 지금부터라도 정해서 유지하는 방법을 택하면 좋을것같습니다.
  - 컴포넌트 props 타이핑 컨벤션

이 모든 방식을 후이가 정하는 것은 무리가 있을 것 같고, 슬랙으로 리모트 소통을 활성화 했으면 좋겠습니다. 4기 프론트분들이 각자가 생각하는 좋은 방식을 적극적으로 제안하는 방식으로 진행하면 좋겠습니다!

**Reply by al-bur** [reply]
괜찮네요. 후이가 혼자 정하기는 어려울 것 같아요!! 프론트끼리 이야기를 해봅시다

**Reply by kwannee** [reply]
오우 좋습니다!! 저도 너무 고맙죠

**Comment by wonsss** [comment]
# 1. 5기가 온보딩 기간에 해당 문서를 보았을 때 부족한 부분이 있다면 피드백 부탁드립니다.

프론트엔드

- 디자인시스템 아이디어 좋아요. 다만, 피그마 외에도 스토리북도 갖고 있으니, 스토리북도 정적 사이트로 자동 배포하면 좋을 것 같아요.(이미 있는데 제가 못 찾은 것일수도?)

# 2. 온보딩 기간에 위의 문서들을 제외하고 어떤 문서가 추가되면 좋을지 아이디어 부탁드립니다.
## 프로젝트 실행 방법 문서화
-  `프로젝트 실행 방법` 문서화가 필요할 것 같습니다. 프론트엔드나 백엔드는 서로의 로컬을 구동하는 방법이 처음에 낯설 수 있고, 프로젝트에 적합한 jdk나 node의 버전을 알지 못하면 막막하기 때문입니다. 근로 초기에 프로젝트 구동 방법부터 막혔던 기억이 있네요. 당시, 프론트 근로 크루들과 정리한 노션 링크를 첨부하오니, 참고하시어 문서화를 해주시면 좋을 것 같습니다.

[자바 버전 변경과 백엔드 서버 구동 방법](https://happysisyphe.notion.site/cee0d2b5e06c46b4bfd5ecf3322656a9)
[노드 버전 변경과 프론트엔드 로컬 서버 구동 방법](https://happysisyphe.notion.site/c57ebce4bd0a4a19aa90cbe0c8b138f8)
[window에서 프론트엔드 로컬 구동 시 이슈](https://happysisyphe.notion.site/window-128ac8c544474d5eb0861df18e025d30)


## 문서화 운영 제안
- [wiki 홈](https://github.com/woowacourse/prolog/wiki)이 아직 비어있는데, 문서화하고 있는 내용을 카테고리별로 분류하여 쉽게 찾아갈 수 있도록 작성하면 어떨까요?

- 회의록 작성 시, wiki와 discussion 중 선택적으로 둘 다 쓰이는 것 같은데, 자주 쓰이고 간편히 작성해야 하는 회의록 특성상 wiki보다는 discussion이 더 나을 것 같아요. 앞으로 discussion으로 일원화했으면 좋겠고, wiki의 기존 회의록 부분은 위에서 말한 카테고리 분류 작업하면서 정리하여 아카이빙하면 좋겠습니다.

- 회의록  제목 네이밍 규칙도 정하면 좋겠어요. 회의록 제목에 대부분 날짜만 적혀 있어서 제목만으로는 내용을 미리 예측하기 어렵네요. 제목에 해당 회의의 `핵심내용`이나 `완성된 기능` 등을 간략히 추가하면 좋을 것 같습니다.
예를 들어, https://github.com/woowacourse/prolog/discussions/1090 회의의 제목은 다음과 같으면 이해하기에 더 좋을 것 같습니다. 
`22.10.24.(월) - 주간회의(이미지 삽입, 로드맵, 권한, 어드민, 자바 버전 변경)`

- 위와 같은 회의록 정리 작업을 위해 크루들끼리 순번을 미리 정해 돌아가면서 서기를 맡으면 좋겠습니다.
